modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -90,253 +92,107 @@ interface WalkResult {
   readonly matchedAttributes: Record<string, AstNode[]>;
 }
 
-const builder = (rootNode: AstNode, html: string, settings: DomParserSettings, schema: Schema, args: ParserArgs) => {
-  const blockElements = extend(makeMap('script,style,head,html,body,title,meta,param'), schema.getBlockElements());
-  const whiteSpaceElements = schema.getWhiteSpaceElements();
-  const nonEmptyElements = schema.getNonEmptyElements();
-  const allWhiteSpaceRegExp = /[ \t\r\n]+/g;
-  const isAllWhiteSpaceRegExp = /^[ \t\r\n]+$/;
-  const startWhiteSpaceRegExp = /^[ \t\r\n]+/;
-  const endWhiteSpaceRegExp = /[ \t\r\n]+$/;
-
-  let node = rootNode;
-  let isInWhiteSpacePreservedElement = Obj.has(whiteSpaceElements, args.context) || Obj.has(whiteSpaceElements, settings.root_name);
-
-  const removeWhitespaceBefore = (node: AstNode): void => {
-    const blockElements = schema.getBlockElements();
-
-    for (let textNode = node.prev; textNode && textNode.type === 3;) {
-      const textVal = textNode.value.replace(endWhiteSpaceRegExp, '');
-
-      // Found a text node with non whitespace then trim that and break
-      if (textVal.length > 0) {
-        textNode.value = textVal;
-        return;
-      }
+const configurePurify = (settings: DomParserSettings, schema: Schema): DOMPurifyI => {
+  const purify = createDompurify();
+  let uid = 0;
+  const config: Config = {
+    RETURN_DOM: true,
+    ALLOW_DATA_ATTR: true,
+    ALLOWED_URI_REGEXP: /^(?:(?:(?:f|ht)tps?|about|blob|file|mailto|tel|callto|cid|xmpp):|[^a-z]|[a-z+.\-]+(?:[^a-z+.\-:]|$))/i,
+  };
 
-      const textNodeNext = textNode.next;
+  if (settings.allow_html_data_urls) {
+    config.ADD_DATA_URI_TAGS = [ 'a', 'img' ];
```

Comment:
This doesn't look right... this open is supposed to allow html data URLs on any element, not just `a` and `img`.

Reply:
This will be fixed via TINY-8202